lib/src/configuration.dart

```diff
@@ -369,8 +369,12 @@ class RealmSchema extends Iterable<SchemaObject> {
 
 /// The signature of a callback that will be invoked if a client reset error occurs for this [Realm].
 ///
-/// Currently, Flexible Sync supports only the [ManualSyncClientResetHandler].
-class SyncClientResetErrorHandler {
+/// A Client Resync is triggered if the device and server cannot agree
+/// on a common shared history for the Realm file,
+/// thus making it impossible for the device to upload or receive any changes.
+/// This can happen if the server is rolled back or restored from backup.
+/// {@category Sync}
+abstract class SyncClientResetErrorHandler {
```

Comment:
```suggestion
abstract class ClientResetHandlerBase {
```

I don't think we need to prefix that with `Sync` because there's no other component that can cause client resets. I also don't think we need the `error` in there because it's somewhat evident.

Reply:
I think having the `Error` there is more clear that it is an error. Actually this will be `ClientResetExceptionHandler` with the rename we are planning to do. Why do we put `Base` on each base class. We could skip that imo. That's not a common pattern.

Reply:
I decided to keep this name and to follow this naming convention, in order to avoid breaking changes.

Reply:
It's common for .NET and I've seen us do it in dart, but am fine with dropping the base. I don't think we need to have exception/error in the name, because it's not necessarily an exceptional situation. You handle a client reset request from the server, which can have multiple reasons to occur.

Reply:
ok lets not have exception or error in the name